Zend/zend_ast.c

```diff
@@ -1722,7 +1722,7 @@ static ZEND_COLD void zend_ast_export_stmt(smart_str *str, zend_ast *ast, int in
 
 	if (ast->kind == ZEND_AST_STMT_LIST ||
 	    ast->kind == ZEND_AST_TRAIT_ADAPTATIONS) {
-		zend_ast_list *list = (zend_ast_list*)ast;
+		const zend_ast_list *list = (const zend_ast_list*)ast;
```

Comment:
Should `zend_ast *ast` not be `const`?

Reply:
The problem is the call to `zend_ast_export_ex()` in the else branch, as this doesn't take a `const` pointer due to it using various `zend_ast_get_*()` functions that don't take a const pointer themselves.

Making all the export functions use const pointers was part of the initial motivation/assumption, as those shouldn't be modifying the AST.

Reply:
Ah I see. What we need essentially is "generic const-ness". That's a little unfortunate, and it might cause more friction between the different APIs. Macros wouldn't help either (nor would `_Generic` I believe), only separate const/non-const getters. Anyway, I don't object. Maybe ask for a 2nd opinion.

Reply:
I *think* C2Y would permit some "generic const-ness" according to https://thephd.dev/improving-_generic-in-c2y

Reply:
This reminds me of the same situation with classes in C++. I'd rather not duplicate getters just to get more const-correctness honestly, it adds a lot of code churn. I'm fine with this patch as-is.

Reply:
@Girgias Ah, that's nice! Sadly, we won't reliable be able to use that for quite a few years. :smiling_face_with_tear: I agree that duplicating getters is not desirable. I'm ok with this as-is, but tbh I would prefer not to merge it unless it improves code-generation.